src/dataecon/DataEcon.jl

```diff
@@ -0,0 +1,562 @@
+# Copyright (c) 2020-2023, Bank of Canada
+# All rights reserved.
+
+module DataEcon
+
+export DEError
+export DEFile, opendaec, closedaec!, truncatedaec
+export root_id, find_fullpath, find_object, delete_object
+export get_attribute, set_attribute, get_fullpath
+export store_scalar, load_scalar
+export store_tseries, load_tseries
+export store_mvtseries, load_mvtseries
+export writedb, write_data
+export readdb, read_data
+
+using ..TimeSeriesEcon
+
+include("C.jl")
+
+#############################################################################
+# open and close daec files
+
+"""
+    struct DEFile ... end
+
+An instance of a *.daec file. Usually there's no need to create instances
+directly. Use [`opendaec`](@ref) and [`closedaec!`](@ref).
+"""
+struct DEFile
+    handle::Ref{C.de_file}
+    fname::String
+end
+
+# forward declarations that need to be imported by I.jl
+function new_catalog end
+function store_scalar end
+function load_scalar end
+function store_tseries end
+function store_mvtseries end
+function load_tseries end
+function load_mvtseries end
+function writedb end
+function readdb end
+function set_attribute end
+function get_attribute end
+
+include("I.jl")
+using .I
+
+const StrOrSym = Union{Symbol,AbstractString}
+
+Base.isopen(de::DEFile) = de.handle[] != C_NULL
+function Base.show(io::IO, de::DEFile)
+    summary(io, de)
+    print(io, ": \"", de.fname, isopen(de) ? "\"" : "\" (closed)")
+end
+Base.unsafe_convert(::Type{C.de_file}, de::DEFile) = isopen(de) ? de.handle[] : throw(ArgumentError("File is closed."))
+
+"""
+    de = opendaec(fname)
+    opendaec(fname) do de
+        ...
+    end
+
+Open the .daec file named in the given `fname` string and return an instance of
+[`DEFile`](@ref). The version with the do-block automatically closes the file.
+Otherwise, call [`closedaec!`](@ref).
+"""
+function opendaec end
+
+function opendaec(fname::AbstractString)
+    handle = Ref{C.de_file}()
+    fname = string(fname)
+    I._check(C.de_open(fname, handle))
+    return DEFile(handle, fname)
+end
+
+function opendaec(f::Function, fname::AbstractString)
+    de = opendaec(fname)
+    try
+        f(de)
+    finally
+        closedaec!(de)
+    end
+end
+
+"""
+    closedaec!(de)
+
+Close a .daec file that was previously opened with [`opendaec`](@ref). The given
+instance of [`DEFile`](@ref) is modified in place to mark it as closed.
+"""
+function closedaec!(de::DEFile)
+    if isopen(de)
+        I._check(C.de_close(de))
+        de.handle[] = C_NULL
+    end
+    return de
+end
+
+"""
+    truncatedaec(de)
+
+Delete all objects in a .daec file. 
+"""
+function truncatedaec end
```

Comment:
> Maybe `cleardaec`?
> 
I went for `truncate` because of Linux https://www.man7.org/linux/man-pages/man1/truncate.1.html, but you're right, that's not a very good name in Julia

> Also, it appears as though this function does nothing.
>
It's not implemented - I just declared a function but didn't define any methods for it yet. 

I think I'll implement this functionality in C as `de_truncate()` and then call it from a new method for `Base.empty!(::DEFile)` in the Julia connector.  What do you think?

Reply:
Done

Reply:
I agree; `Base.empty!` makes sense.